### PR TITLE
Stop stringifying true values in h

### PR DESCRIPTION
**Pull request by jxodwyer**
# Fix Issue https://github.com/developit/preact/issues/433
It has been requested that we would no longer want to render true bool values. Similar to that of functionality seen in React.

##### Input:
`<div>{true}</div>`
##### Output:
`<div>true</div>`

##### Input:
`<div>{true}</div>`
##### Output:
`<div></div>`

### CODE REVIEW

**Comment by developit** [comment]
Nice! Was toying with this, and I like your solution.

**Comment by jxodwyer** [comment]
Whoa that was fast! Thanks a lot @developit.

File: src/h.js [hunk 1]
```diff
@@ -28,8 +28,8 @@ export function h(nodeName, attributes) {
 		if ((child = stack.pop()) instanceof Array) {
 			for (i=child.length; i--; ) stack.push(child[i]);
 		}
-		else if (child!=null && child!==false) {
-			if (typeof child=='number' || child===true) child = String(child);
+		else if (child!=null && child!==true && child!==false) {
```
Comment (NekR) [inline src/h.js]: `typeof child != 'boolean'` maybe?

**Comment by jxodwyer** [comment]
@NekR nice catch. Just updated

**Comment by developit** [comment]
@jxodwyer @NekR strangely, the `!='boolean'` check is 4 bytes bigger after gzip. Odd! Maybe do the verbose one?

**Comment by jxodwyer** [comment]
@developit that's super odd. Just reverted it back

File: src/h.js (hunk 1 above)
Comment (noomorph) [inline src/h.js]: @NekR , @developit, isn't it slower? There's a small difference between these two ways in Chrome (in favor of typeof), but Firefox, for example, is 2x faster with plain comparison (`=== true || === false`). 

https://esbench.com/bench/58de1a6299634800a0347ba9

Comment (developit) [inline src/h.js]: I ended up opting to side with gzipped byte size here, which favors the explicit `true` / `false` check.

**Pull request merged**